### PR TITLE
 add optional error handler for c++ api. #3243 

### PR DESCRIPTION
**Pull request by FONQRI**
Hello
I add sign-off in commits as you sed in https://github.com/gnuradio/gnuradio/pull/3243

### CODE REVIEW

File: gnuradio-runtime/include/gnuradio/top_block.h [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 #include <gnuradio/api.h>
 #include <gnuradio/hier_block2.h>
+#include <gnuradio/thread/basic_error_handler.h>
```
Comment (mbr0wn) [inline gnuradio-runtime/include/gnuradio/top_block.h:16]: This requires updating the bindings for this file.

**Review by mbr0wn — CHANGES_REQUESTED** [review]
Can you please fix the formatting (using clang-format) and the Python bindings. You should be able to tell if the build works on your machine for tesing.

**Comment by mbr0wn** [comment]
Oh, and it needs a rebase.

**Comment by FONQRI** [comment]
> Oh, and it needs a rebase.
Yes but i'm a little busy for now i'll do it later.

File: gnuradio-runtime/include/gnuradio/thread/thread_body_wrapper.h [hunk 2]
```diff
@@ -67,7 +87,8 @@ class thread_body_wrapper
         }
     }
 };
-
+template <class F>
+std::shared_ptr<basic_error_handler> thread_body_wrapper<F>::errorHandler;
```
Comment (mbr0wn) [inline gnuradio-runtime/include/gnuradio/thread/thread_body_wrapper.h:91]: The camelCase also seems wrong here. We usually use snake_case.

File: gnuradio-runtime/include/gnuradio/thread/basic_error_handler.h [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#ifndef BASIC_ERROR_HANDLER_H
```
Comment (mbr0wn) [inline gnuradio-runtime/include/gnuradio/thread/basic_error_handler.h:1]: This file needs a header.

**Review by mbr0wn — CHANGES_REQUESTED** [review]

**Comment by mbr0wn** [comment]
Closing due to lack of action.

**Comment by mbr0wn** [comment]
You can reopen of course upon resolution.

File: gnuradio-runtime/include/gnuradio/top_block.h (hunk 1 above)
Comment (FONQRI) [reply gnuradio-runtime/include/gnuradio/top_block.h:16]: hi
it's a long time.
can you help me with update bindings?
I created a new PR https://github.com/gnuradio/gnuradio/pull/4795

**Pull request closed without merging**